pkgs/applications/networking/mailreaders/mblaze/default.nix

```diff
@@ -1,31 +1,54 @@
-{ stdenv, lib, fetchFromGitHub, fetchpatch, libiconv, ruby ? null }:
+{ coreutils, fetchFromGitHub, fetchpatch, file, gawk, gnugrep, gnused, less
+, lib, libiconv, makeWrapper, nano, stdenv, ruby ? null
+}:
 
 stdenv.mkDerivation rec {
   pname = "mblaze";
-  version = "0.5.1";
+  version = "0.7";
 
   buildInputs = [ ruby ] ++ lib.optionals stdenv.isDarwin [ libiconv ];
 
+  nativeBuildInputs = [ makeWrapper ];
+
   src = fetchFromGitHub {
     owner = "chneukirchen";
     repo = "mblaze";
     rev = "v${version}";
-    sha256 = "11x548dl2jy9cmgsakqrzfdq166whhk4ja7zkiaxrapkjmkf6pbh";
+    sha256 = "11cr26ac80y41a9s63qhab8wj5hq8ydsd9b142b2pdr8ii2f15yw";
   };
 
-  patches = [
-    (fetchpatch {
-      url = "https://github.com/leahneukirchen/mblaze/commit/53151f4f890f302291eb8d3375dec4f8ecb66ed7.patch";
-      sha256 = "1mcyrh053iiyzdhgm09g5h3a77np496whnc7jr4agpk1nkbcpfxc";
-    })
-  ];
+  # patches = [
+  #   (fetchpatch {
+  #     url = "https://github.com/leahneukirchen/mblaze/commit/53151f4f890f302291eb8d3375dec4f8ecb66ed7.patch";
+  #     sha256 = "1mcyrh053iiyzdhgm09g5h3a77np496whnc7jr4agpk1nkbcpfxc";
+  #   })
+  # ];
 
   makeFlags = [ "PREFIX=$(out)" ];
 
   postInstall = ''
     install -Dm644 -t $out/share/zsh/site-functions contrib/_mblaze
   '' + lib.optionalString (ruby != null) ''
     install -Dt $out/bin contrib/msuck contrib/mblow
+
+    # The following wrappings are used to preserve the executable
+    # names (the value of $0 in a script). The script mcom is
+    # designed to be run directly or via symlinks such as mrep. Using
+    # symlinks changes the value of $0 in the script, and makes it
+    # behave differently. When using the wrapProgram tool, the resulting
+    # wrapper breaks this behaviour. The following wrappers preserve it.
```

Comment:
Interesting! The argv0 option to wrapProgram doesn’t fix this?

Reply:
I've learned it doesn't work for scripts with shebangs. It does if you remove the shebang, but then you have to explicitly pass it to bash everytime you want to run it…